cmd/main.go

```diff
@@ -63,8 +71,19 @@ func main() {
 		isCLI = true
 	}
 	util.SetAutoMaxProcs(isCLI)
+	if isCLI {
+		command.SilenceErrors = true
+		command.SilenceUsage = true
+	}
 
 	if err := command.Execute(); err != nil {
+		if isCLI {
+			if pluginErr := cli.HandlePluginCommand(o.PluginHandler, o.Arguments[1:], 1); pluginErr != nil {
+				fmt.Fprintf(os.Stderr, "Error: %v\n", pluginErr)
+				os.Exit(1)
```

Comment:
I am afraid that this is going to swallow the error returned by the `command.Execute()` execution above.
Consider the scenario:
1. run `argocd app list`
2. `argocd app list` returns an error
3. `cli.HandlePluginCommand` is invoked and will always return error `command not found`
4. exit(1)
5. the main `argocd app list` error is never presented to the user

Reply:
yes this is correct, i didn't think about this scenairo in depth. Other way to do is we can prioritize the execute of a plugin over the normal argocd command. WDYT of it?

(Basically, I'm trying to find all ways possible before introducing the plugin subcommand).

Reply:
In addition to what @leoluz said, it seems like that even if the plugin command executes successfully, the output would _still_ fail (because it doesn't handle the case where it should exit successfully - instead it seems like it would go to line 86 and fail with an error anyway).

Reply:
@nitishfy please make sure that those error cases are properly handled. Preferably covered by unit tests.